jujugui/static/gui/src/app/components/status/status.js

```diff
@@ -4,172 +4,199 @@
+      case 'idle':
+      case 'started':
+        return 'ok';
+      case 'blocked':
+      case 'down':
+      case 'error':
+        return 'error';
+    }
+    return '';
+  }
 
   /**
-    Render the component.
+    Sort by the given index.
+    @param index {Int} The index to sort by.
+    @param {Object} a The first value.
+    @param {Object} b The second value.
+    @returns {Array} The sorted array.
   */
-  render() {
-    return (
-      <juju.components.Panel
-        instanceName="status-view"
-        visible={true}>
-        <div className="status-view__content">
-          {this.renderStatus()}
-        </div>
-      </juju.components.Panel>
-    );
+  _byIndex(index, a, b) {
+    if (a[index] < b[index])
+      return -1;
+    if (a[index] > b[index])
+      return 1;
+    return 0;
   }
 
   /**
-    Render the current model status.
+    Generate the current model status.
     @returns {Object} The resulting element.
   */
-  renderStatus() {
+  _generateStatus() {
     const elements = [];
     const db = this.props.db;
     const model = this.props.model;
     if (!model.environmentName) {
       // No need to go further: we are not connected to a model.
       return 'Cannot show the status: the GUI is not connected to a model.';
     }
-    elements.push(this._renderModel(model));
+    elements.push(this._generateModel(model));
     if (db.remoteServices.size()) {
-      elements.push(this._renderRemoteApplications(db.remoteServices));
+      elements.push(this._generateRemoteApplications(db.remoteServices));
     }
     if (db.services.size()) {
       elements.push(
-        this._renderApplications(db.services),
-        this._renderUnits(db.services)
+        this._generateApplications(db.services),
+        this._generateUnits(db.services)
       );
     }
     if (db.machines.size()) {
-      elements.push(this._renderMachines(db.machines));
+      elements.push(this._generateMachines(db.machines));
     }
     if (db.relations.size()) {
-      elements.push(this._renderRelations(db.relations));
+      elements.push(this._generateRelations(db.relations));
     }
     return elements;
   }
 
   /**
-    Render the model fragment of the status.
+    Generate the model fragment of the status.
     @param {Object} model The model attributes.
     @returns {Object} The resulting element.
   */
-  _renderModel(model) {
+  _generateModel(model) {
     return (
-      <table key="model">
-        <thead>
-          <tr>
-            <th>Model</th>
-            <th>Cloud/Region</th>
-            <th>Version</th>
-            <th>SLA</th>
-          </tr>
-        </thead>
-        <tbody>
-          <tr key={model.environmentName}>
-            <td>{model.environmentName}</td>
-            <td>{model.cloud}/{model.region}</td>
-            <td>{model.version}</td>
-            <td>{model.sla}</td>
-          </tr>
-        </tbody>
-      </table>
-    );
+      <juju.components.BasicTable
+        columns={[{
```

Comment:
The reason I was using columns and not something like headers is that the columns definition is also used to shape the content as well (it sets the size and classes for the row columns as well as the header). Does that help or would you like to see another name?